models/issue.go

```diff
@@ -827,9 +827,11 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 			if err != nil {
 				return fmt.Errorf("hasAccess [user_id: %d, repo_id: %d]: %v", assigneeID, opts.Repo.ID, err)
 			}
-			if valid {
-				validAssigneeIDs = append(validAssigneeIDs, assigneeID)
+			if !valid {
+				return ErrUserDoesNotHaveAccessToRepo{UserID: assigneeID, RepoName: opts.Repo.Name}
 			}
+
+			validAssigneeIDs = append(validAssigneeIDs, assigneeID)
```

Comment:
Please remove this slice since it is unused

Reply:
Done.